device/globals/item.gd

```diff
@@ -282,10 +288,24 @@ func _update_terrain():
 	var color = terrain.get_terrain(pos)
 	var scale_range = terrain.get_scale_range(color.b)
 
+	# The item's - as the player's - `animations` define the direction
+	# as 1 or -1. This is stored as `pose_scale` and the easiest way
+	# to flip a node is multiply its x-axis scale.
+	scale_range.x *= pose_scale
+
 	if scale_on_map && (self is Node2D) && scale_range != get_scale():
-		var c = terrain.get_terrain(pos)
-		var s = terrain.get_scale_range(c.b)
-		set_scale(s)
+		# Check children to see if `interact_pos` is a child of ours, and if so,
+		# take a backup of the global position, because it will be affected by scaling.
+		var interact_global_position
+		if interact_pos in get_children():
+			interact_global_position = interact_pos.get_global_position()
```

Comment:
We're already saving a reference to `interact_pos` at the bottom of `_ready()`, if one is present. All current code should access the interact position via the getter function. Are there cases where external consumers of that function might get the wrong, flipped position?

Reply:
No, probably not.

Reply:
`has_node("interact_pos")` does seem to work just as well, force-pushed it. Thanks for the tip!